packages/ringcentral-widgets/containers/CallCtrlPage/index.js

```diff
@@ -125,46 +140,131 @@ class CallCtrlPage extends Component {
+          if (this.checkCalleeType(lastCall) === calleeTypes.contacts) {
+            const lastTo = {
+              avatarUrl: lastCall.toMatches[0].profileImageUrl,
+              name: lastCall.toName,
+              status: lastCall.webphoneSession.callStatus,
+              calleeType: calleeTypes.contacts,
+              sessionId: lastCall.webphoneSession.id,
+              session: lastCall.webphoneSession
+            };
+            this.setState(prev => ({
+              ...prev,
+              lastTo
+            }));
+            // to fetch avatarurl again if the profileimageurl is null
+            if (!lastCall.toMatches[0].profileImageUrl) {
+              this.props.getAvatarUrl(lastCall.toMatches[0]).then((avatarUrl) => {
+                this.setState(prev => ({
+                  ...prev,
+                  lastTo: {
+                    ...prev.lastTo,
+                    avatarUrl
+                  }
+                }));
+              });
             }
-          });
+          } else if (this.checkCalleeType(lastCall) === calleeTypes.unknow) {
+            this.setState(prev => ({
+              ...prev,
+              lastTo: {
+                calleeType: calleeTypes.unknow,
+                avatarUrl: null,
+                sessionId: lastCall.webphoneSession ? lastCall.webphoneSession.id : null,
+                status: lastCall.webphoneSession ? lastCall.webphoneSession.callStatus : null,
+                name: lastCall.to.phoneNumber
+              }
+            }));
+          }
         }
       }
     } else {
-      const { conferencePartiesAvatarUrls } = this.props;
-      this.setState({
+      const { conferencePartiesAvatarUrls, conferenceCall } = this.props;
+      if (Object.values(conferenceCall.conferences).length) {
+        this.lastToListen(Object.values(conferenceCall.conferences)[0].session.id);
+        this.setState(() => ({
+          lastTo: {
+            calleeType: calleeTypes.conference,
+            avatarUrl: conferencePartiesAvatarUrls[0],
+            extraNum: conferencePartiesAvatarUrls.length - 1,
+            sessionId: Object.values(conferenceCall.conferences)[0].session.id
+          }
+        }));
+      }
+    }
+  }
+  updateLastToState(nextProps) {
+    if (Object.keys(nextProps.conferenceCall.conferences).length) {
+      const part = this.props.getPartyProfiles().length - 1;
+      if (!this._mounted) {
+        return;
+      } else {
+        this.setState(prev => ({
+          lastTo: {
+            ...prev.lastTo,
+            extraNum: part
+          }
+        }));
+      }
+    }
+  }
+  terminatedCallback(session, isLastTo = false) {
+    return () => this.terminatedSuccessHander(session, isLastTo);
+  }
+  terminatedSuccessHander(session, isLastTo) {
+    session.removeListener('terminated', this.terminatedSuccessHander);
+    const { routerInteraction, webphone, conferenceCall } = this.props;
+    if (isLastTo) {
```

Comment:
check the component mount status when calling the setState on the method, because it's discrete